app/experimenter/nimbus-ui/src/components/Summary/EndExperiment/index.tsx

```diff
@@ -0,0 +1,126 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { useMutation } from "@apollo/client";
+import React, { useCallback, useState } from "react";
+import { Alert, Button } from "react-bootstrap";
+import { END_EXPERIMENT_MUTATION } from "../../../gql/experiments";
+import { SUBMIT_ERROR } from "../../../lib/constants";
+import { getStatus } from "../../../lib/experiment";
+import { endExperiment_endExperiment as EndExperimentResult } from "../../../types/endExperiment";
+import { getExperiment_experimentBySlug } from "../../../types/getExperiment";
+import { ExperimentIdInput } from "../../../types/globalTypes";
+
+const EndExperiment = ({
+  experiment,
+  status,
+}: {
+  experiment: getExperiment_experimentBySlug;
+  status: ReturnType<typeof getStatus>;
+}) => {
+  const [submitError, setSubmitError] = useState<string | null>(null);
+  const [showEndConfirmation, setShowEndConfirmation] = useState(false);
+  const [endRequested, setEndRequested] = useState(false);
+  const isEnding = status.ending || endRequested;
+
+  const [endExperiment, { loading: endExperimentLoading }] = useMutation<
+    { endExperiment: EndExperimentResult },
+    { input: ExperimentIdInput }
+  >(END_EXPERIMENT_MUTATION);
+
+  const toggleShowEndConfirmation = useCallback(
+    () => setShowEndConfirmation(!showEndConfirmation),
+    [showEndConfirmation, setShowEndConfirmation],
+  );
+
+  const onConfirmEndClicked = useCallback(async () => {
+    try {
+      const result = await endExperiment({
```

Comment:
Do we maybe need a "loading" state here while this resolves? Probably not a big deal since this mutation should be idempotent

Reply:
Both buttons are [disabled](https://github.com/mozilla/experimenter/pull/4545/files#diff-a34086e9954d0954013483aa493baa2be3f4c9f94eadf5ca88198813109060daR97) while the mutation does its thing - that what you mean?

Reply:
Oh, derp, I missed that!